README.md

```diff
@@ -28,38 +28,53 @@ This driver is built using [Flutter Integration Test](https://docs.flutter.dev/c
 ## How to Use Appium Flutter Integration Driver
 
 1. In your Flutter app's `pubspec.yaml`, add the following dependencies:
-   
+
    Get the latest version from `https://pub.dev/packages/appium_flutter_server/install`
 
-    ```yaml
-    dev_dependencies:
-      appium_flutter_server: ^0.0.4
-    ```
+   ```yaml
+   dev_dependencies:
+     appium_flutter_server: ^0.0.4
+   ```
 
 2. Create a directory called `integration_test` in the root of your Flutter project.
 3. Create a file called `appium_test.dart` in the `integration_test` directory.
 4. Add the following code to the `appium_test.dart` file:
 
-    ```dart
-    import 'package:appium_flutter_server/appium_flutter_server.dart';
-    import 'package:appium_testing_app/main.dart';
-
-    void main() {
-      initializeTest(app: const MyApp());
-    }
-    ```
+   ```dart
+   import 'package:appium_flutter_server/appium_flutter_server.dart';
+   import 'package:appium_testing_app/main.dart';
+
+   void main() {
+     initializeTest(app: const MyApp());
+   }
+   ```
+   If you are in need to configure certain prerequists before the testing app is loaded, you can try the following code:
+   ```dart
+   import 'package:appium_testing_app/main.dart'; as app;
+   void main() {
+     initializeTest(
+       callback: (WidgetTester tester) async {
+         injector.configurePreRequisites(
+           completion: () async {
+             await tester.pumpWidget(const app.MyApp());
+           },
+         );
+       },
+     );
+   }
+   ```
```

Comment:
Can we slightly modify the example here to keep it simple for users.

If you are in need to configure certain prerequisites before the testing app is loaded, you can pass a callback function which will be invoked once the test is initialised. You can add any custom logics before creating the app object.

```dart
import 'package:appium_testing_app/main.dart'; as app;
void main() {
  initializeTest(
    callback: (WidgetTester tester) async {
      // Perform any prerequisite steps or intialise any dependencies required by the app
      // and make sure to pump the app widget using below statement.
      await tester.pumpWidget(const app.MyApp());
    },
  );
}
```